test/e2e/network/service.go

```diff
@@ -1971,6 +1972,88 @@ var _ = SIGDescribe("Services", func() {
 		framework.ExpectNoError(err, "failed to create service: %s in namespace: %s", serviceName, ns)
 	})
 
+	ginkgo.It("should not allocate NodePorts for Type LoadBalancer if allocateLoadBalancerNodePorts=false [Feature:ServiceLBNodePortControl]", func() {
+		serviceName := "lb-disable-nodeport"
+		ns := f.Namespace.Name
+
+		t := NewServerTest(cs, ns, serviceName)
+		defer func() {
+			defer ginkgo.GinkgoRecover()
+			errs := t.Cleanup()
+			if len(errs) != 0 {
+				framework.Failf("errors in cleanup: %v", errs)
+			}
+		}()
+
+		service := t.BuildServiceSpec()
+		service.Spec.Type = v1.ServiceTypeLoadBalancer
+		service.Spec.AllocateLoadBalancerNodePorts = utilpointer.BoolPtr(false)
+
+		ginkgo.By("creating service " + serviceName + " with type LoadBalancer and no node ports in namespace " + ns)
+		service, err := t.CreateService(service)
+		framework.ExpectNoError(err, "failed to create service: %s in namespace: %s", serviceName, ns)
+
+		foundNodePorts := false
+		for _, port := range service.Spec.Ports {
+			if port.NodePort > 0 {
+				foundNodePorts = true
+			}
+		}
+
+		if foundNodePorts {
+			framework.Failf("service %s/%s had unexpected node ports allocated", ns, serviceName)
+		}
+	})
+
+	ginkgo.It("should not de-allocate NodePorts for LoadBalancer if allocateLoadBalancerNodePorts switches from true to false [Feature:ServiceLBNodePortControl]", func() {
+		serviceName := "lb-disable-nodeport"
+		ns := f.Namespace.Name
+
+		t := NewServerTest(cs, ns, serviceName)
+		defer func() {
+			defer ginkgo.GinkgoRecover()
+			errs := t.Cleanup()
+			if len(errs) != 0 {
+				framework.Failf("errors in cleanup: %v", errs)
+			}
+		}()
```

Comment:
you know why is recommended the ServiceTestJig over this fixture?
https://github.com/kubernetes/kubernetes/blob/6dddea5abfa20a22e93114d66735441f04f4a39c/test/e2e/network/service.go#L1844-L1845
This seems to come from the cleanup from Tim
https://github.com/kubernetes/kubernetes/commit/b7782e73b648c8d866287ff5cc19a6efd1164170

Reply:
I was hoping TestJig would be used EVERYWHERE and streamline the tests by taking (and hiding) a lot of boilerplate decisions.  I obviously didn't finish :(